ts/components/Pinpad/index.tsx

```diff
@@ -31,6 +32,7 @@ interface Props {
   onPinResetHandler?: () => void;
   onFingerPrintReq?: () => void;
   onDeleteLastDigit?: () => void;
+  remainingAttempts?: number;
```

Comment:
could it be `Option<number>`?
In this way in your code you can use _map_ or _fold_ instead of write checks or convert it using _fromNullable_

Reply:
The problem with `Option<number>` in the props for a reusable component as the pinpad is that if I want to use this component without the optionals `remainingAttempts`, I will be forced to specify it in the props with `remainingAttempts={none}`.

Reply:
This is a good point I didn't spot. As you told option props allow to don't specify that prop if you don't need it in that component